src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/inplace_resolve_io.cpp

```diff
@@ -0,0 +1,321 @@
+        init_input_shapes({input_shape});
+        ov::ParameterVector params;
+        for (auto&& shape : inputDynamicShapes) {
+            params.push_back(std::make_shared<ov::op::v0::Parameter>(precision, shape));
+        }
+        auto soft_max = std::make_shared<ov::op::v1::Softmax>(params.front(), softmax_axis);
+
+        auto add_const = ov::op::v0::Constant::create(precision, {1}, std::vector<float>({1.0f}));
+        auto add_0 = ov::test::utils::make_eltwise(soft_max, add_const, ov::test::utils::EltwiseTypes::ADD);
+        auto result_0 = std::make_shared<ov::op::v0::Result>(add_0);    // dummy output
+
+        auto reshape_param_0 = ov::op::v0::Constant::create(ov::element::i32, {1}, {0});
+        auto reshape_0 = std::make_shared<ov::op::v0::Unsqueeze>(soft_max, reshape_param_0);
+        reshape_0->set_friendly_name("reshape_0");
+        auto result_2 = std::make_shared<ov::op::v0::Result>(reshape_0);    // dummy output
+
+        auto reshape_param_1 = ov::op::v0::Constant::create(ov::element::i32, {1}, {0});
+        auto reshape_1 = std::make_shared<ov::op::v0::Unsqueeze>(reshape_0, reshape_param_1);
+        reshape_1->set_friendly_name("reshape_1");
+        auto result_1 = std::make_shared<ov::op::v0::Result>(reshape_1);    // target output
+
+        function = std::make_shared<ov::Model>(ov::ResultVector{result_0, result_1, result_2}, params, "Subgraph2");
+    }
+};
+
+TEST_P(SoftmaxAddReshapeTwoOutputsSubgraphTest, smoke_CompareWithRefs) {
+    run();
+}
+
+class InputReshapeOutputSubgraphTest : public InplaceResolveIOTestBase {
+/*This test runs the following subgraph:
+
+                      param
+                        |
+                        |
+                    Reshape0
+                        |
+                        |
+                     Result0
+
+Edge Reshape0 -> Result0 cannot be referenced by its upstreams as its upstream is an input.
+*/
+protected:
+    void SetUp() override {
+        targetDevice = ov::test::utils::DEVICE_CPU;
+        auto& input_shape = this->GetParam();
+        init_input_shapes({input_shape});
+        ov::ParameterVector params;
+        for (auto&& shape : inputDynamicShapes) {
+            params.push_back(std::make_shared<ov::op::v0::Parameter>(precision, shape));
+        }
+
+        auto reshape_param_0 = ov::op::v0::Constant::create(ov::element::i32, {1}, {0});
+        auto reshape_0 = std::make_shared<ov::op::v0::Unsqueeze>(params.front(), reshape_param_0);
+        auto result_0 = std::make_shared<ov::op::v0::Result>(reshape_0);    // target output
+
+        function = std::make_shared<ov::Model>(ov::ResultVector{result_0}, params, "Subgraph3");
+    }
+};
+
+TEST_P(InputReshapeOutputSubgraphTest, smoke_CompareWithRefs) {
+    run();
+}
+
+
+namespace {
+
+const std::vector<std::vector<InputShape>> inputShapes = {
+    {
+        // static
+        {{2, 64}, {{2, 64}}}
+    },
+    {
+        // dynamic
+        {{2, -1}, {{2, 64}}}
```

Comment:
Could you please add at least one more target shape to the dynamic case?

Reply:
done